### PR TITLE
feat: add ledger signing to smart account module

### PR DESCRIPTION
**Pull request by PaddyMc**
## What is the purpose of the change

The smart account `AddAuthenticator` and `RemoveAuthenticator` messages were missing the correct functions to satisfy the legacy message interface, thus ledger signing couldn't be used.

Also some weirdness around having a Type for the message and also needed to define a Type() function so I need to change that.

Sister PR here => https://github.com/osmosis-labs/osmosis-frontend/pull/3656

## Testing and Verifying

- `cd x/smart-account`
- `go test ./...`

### CODE REVIEW

File: proto/osmosis/smartaccount/v1beta1/tx.proto [hunk 1]
```diff
@@ -34,7 +34,7 @@ message MsgAddAuthenticatorResponse { bool success = 1; }
 // MsgRemoveAuthenticatorRequest defines the Msg/RemoveAuthenticator request
 // type.
 message MsgRemoveAuthenticator {
-  option (amino.name) = "osmosis/MsgRemoveAuthenticator";
+  option (amino.name) = "osmosis/smartaccount/remove-authenticator";
```
Comment (czarcas7ic) [inline proto/osmosis/smartaccount/v1beta1/tx.proto:37]: did we verify name length on this? sometimes ledger can only handle certain lengths

**Review by czarcas7ic — APPROVED** [review]
High level ACK, we of course should test on v26 testnet prior to gov prop. Very excited for this!

**Review by ValarDragon — APPROVED** [review]
Nice

**Comment by ValarDragon** [comment]
Lets test that the character count for each of these names is sub-40 chars (IIRC thats the limit)

File: proto/osmosis/smartaccount/v1beta1/tx.proto (hunk 1 above)
Comment (PaddyMc) [reply proto/osmosis/smartaccount/v1beta1/tx.proto:37]: Selection_063

**Comment by PaddyMc** [comment]
> Lets test that the character count for each of these names is sub-40 chars (IIRC thats the limit)

Selection_063

File: proto/osmosis/smartaccount/v1beta1/tx.proto (hunk 1 above)
Comment (PaddyMc) [reply proto/osmosis/smartaccount/v1beta1/tx.proto:37]: I signed with a ledger to make sure this works as well => https://osmosis-network.slack.com/archives/C027ELA290B/p1722618905776139?thread_ts=1722618892.008519&cid=C027ELA290B

**Pull request merged**